p2p_beef_tx.go

```diff
@@ -91,17 +88,29 @@ func decodeBUMPs(beefBytes []byte) ([]BUMP, []byte, error) {
 		return nil, nil, errors.New("cannot decode BUMP - no bytes provided")
 	}
 
-	bumps := make([]BUMP, 0)
 	nBump, bytesUsed := bt.NewVarIntFromBytes(beefBytes)
+
+	if nBump == 0 {
+		return nil, nil, errors.New("invalid BEEF- lack of BUMPs")
+	}
+
 	beefBytes = beefBytes[bytesUsed:]
 
+	bumps := make([]BUMP, 0, int(nBump))
 	for i := 0; i < int(nBump); i++ {
 		if len(beefBytes) == 0 {
 			return nil, nil, errors.New("insufficient bytes to extract BUMP blockHeight")
 		}
 		blockHeight, bytesUsed := bt.NewVarIntFromBytes(beefBytes)
 		beefBytes = beefBytes[bytesUsed:]
-		bumpPaths, remainingBytes, err := decodeBUMPPathsFromStream(beefBytes)
+
+		treeHeight := beefBytes[0]
+		if int(treeHeight) > maxTreeHeight {
```

Comment:
Do we cover this path in unit tests? I think that we should. I mean when this error occurs.

Reply:
New test case in tests